From 1feedf7875d4e66167c20975ac349ca4d08233d3 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Mon, 5 Apr 2021 17:22:50 +0300
Subject: [PATCH] arm64: dts: imx8mp: move to kernel 5.10 based release

Move to NXP 5.10 1.0.0 based release.
Fixed two issues -
1. HDMI device tree
2. PCIe clock tree (still not functional)

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 .../freescale/imx8mp-hummingboard-pulse.dts   | 43 ++++++++-----------
 1 file changed, 19 insertions(+), 24 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dts b/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dts
index 5f9b24547c50..a6c120027271 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dts
@@ -438,11 +438,7 @@ &irqsteer_hdmi {
 	status = "okay";
 };
 
-&hdmimix_clk {
-	status = "okay";
-};
-
-&hdmimix_reset {
+&hdmi_blk_ctrl {
 	status = "okay";
 };
 
@@ -506,22 +502,22 @@ &micfil {
 };
 
 &pcie{
-        pinctrl-names = "default";
-        pinctrl-0 = <&pinctrl_pcie>;
-        disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
-        reset-gpio = <&gpio4 28 GPIO_ACTIVE_LOW>;/*need to verify which igpio is final*/
-        ext_osc = <0>;
-        clocks = <&clk IMX8MP_CLK_HSIO_AXI_DIV>,
-                 <&clk IMX8MP_CLK_PCIE_AUX>,
-                 <&clk IMX8MP_CLK_PCIE_PHY>,
-                 <&clk IMX8MP_CLK_PCIE_ROOT>;
-        clock-names = "pcie", "pcie_aux", "pcie_phy", "pcie_bus";
-        assigned-clocks = <&clk IMX8MP_CLK_HSIO_AXI_SRC>,
-                          <&clk IMX8MP_CLK_PCIE_AUX>;
-        assigned-clock-parents = <&clk IMX8MP_SYS_PLL2_500M>,
-                                 <&clk IMX8MP_SYS_PLL2_50M>;
-        reserved-region = <&rpmsg_reserved>;
-        status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcie>;
+	disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio4 28 GPIO_ACTIVE_LOW>;/*need to verify which igpio is final*/
+	ext_osc = <0>;
+	clocks = <&clk IMX8MP_CLK_HSIO_ROOT>,
+		 <&clk IMX8MP_CLK_PCIE_AUX>,
+		 <&clk IMX8MP_CLK_HSIO_AXI>,
+		 <&clk IMX8MP_CLK_PCIE_ROOT>;
+	clock-names = "pcie", "pcie_aux", "pcie_phy", "pcie_bus";
+	assigned-clocks = <&clk IMX8MP_CLK_HSIO_AXI>,
+			  <&clk IMX8MP_CLK_PCIE_AUX>;
+	assigned-clock-rates = <500000000>, <10000000>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL2_500M>,
+				 <&clk IMX8MP_SYS_PLL2_50M>;
+	status = "okay";
 };
 
 &pcie_phy{
@@ -535,8 +531,8 @@ &sai3 {
 	assigned-clocks = <&clk IMX8MP_CLK_SAI3>;
 	assigned-clock-parents = <&clk IMX8MP_AUDIO_PLL1_OUT>;
 	assigned-clock-rates = <12288000>;
-	clocks = <&audiomix_clk IMX8MP_CLK_AUDIOMIX_SAI3_IPG>, <&clk IMX8MP_CLK_DUMMY>,
-		 <&audiomix_clk IMX8MP_CLK_AUDIOMIX_SAI3_MCLK1>, <&clk IMX8MP_CLK_DUMMY>,
+	clocks = <&audio_blk_ctrl IMX8MP_CLK_AUDIO_BLK_CTRL_SAI3_IPG>, <&clk IMX8MP_CLK_DUMMY>,
+		 <&audio_blk_ctrl IMX8MP_CLK_AUDIO_BLK_CTRL_SAI3_MCLK1>, <&clk IMX8MP_CLK_DUMMY>,
 		 <&clk IMX8MP_CLK_DUMMY>;
 	clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
 	fsl,sai-mclk-direction-output;
@@ -799,7 +795,6 @@ MX8MP_IOMUXC_GPIO1_IO10__GPIO1_IO10		0x19
 			MX8MP_IOMUXC_ECSPI2_SS0__GPIO5_IO13		0x41
 			MX8MP_IOMUXC_NAND_DATA02__GPIO3_IO08		0x41
 			MX8MP_IOMUXC_SAI3_RXFS__GPIO4_IO28		0x41/*pcie reset*/
-			//MX8MP_IOMUXC_GPIO1_IO00__GPIO1_IO00		0x41/*pcie reset*/
 		>;
 	};
 
-- 
2.25.1

